src/Elmah/XmlFileErrorLog.cs

```diff
@@ -158,10 +173,36 @@ public override string Log(Error error)
                 writer.Flush();
             }
 
+            DeleteOldFiles();
             return errorId;
         }
 
         /// <summary>
+        /// Deletes Old files according the <code>size</code> parameter
+        /// </summary>
+        private void DeleteOldFiles()
+        {
+            try
+            {
+                if (!Directory.Exists(LogPath))
+                    return;
+
+                DirectoryInfo dir = new DirectoryInfo(LogPath);
+                FileInfo[] logFiles = dir.GetFiles("error*.xml");
```

Comment:
File enumeration should be reconciled with [that in `GetErrors`](https://github.com/elmah/Elmah/blob/dd67341a96406e66f107480d66fd6733fa07dab8/src/Elmah/XmlFileErrorLog.cs#L183-L187).